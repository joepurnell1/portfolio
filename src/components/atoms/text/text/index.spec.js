import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import Text from './index';

const theme = {
  Global: {
    textColour: '#DA221E',
  },
};

describe('Text component', () => {
  it('should render default as expected', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Text>Blurred vision and the hobble of thick prose</Text>
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render with custom colour as expected', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Text colour="#F00B42">Why empower misogyny while violence towards women grows?</Text>
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
