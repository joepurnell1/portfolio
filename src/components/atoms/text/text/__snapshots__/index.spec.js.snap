// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should render default as expected 1`] = `
<p
  className="sc-bdVaJa kztLhk"
>
  Blurred vision and the hobble of thick prose
</p>
`;

exports[`Text component should render with custom colour as expected 1`] = `
<p
  className="sc-bdVaJa kztLhk"
>
  Why empower misogyny while violence towards women grows?
</p>
`;
